### PR TITLE
New versions 3.4.1 and 3.5.0

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added support for 3.4.1, 3.5.0
- removed support for 3.2.1, 3.3.0
- history server now has metrics via jmx exporter
- fixed tests (spark examples, [ny-tlc](https://github.com/stackabletech/ny-tlc-report/pull/6))
- adapted docs

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: tests/templates/kuttl/iceberg/10-deploy-spark-app.yaml.j2 [hunk 1]
```diff
@@ -45,7 +45,8 @@ spec:
           mountPath: /stackable/spark/jobs
   deps:
     packages:
-      - org.apache.iceberg:iceberg-spark-runtime-{{ test_scenario['values']['spark'].rstrip('.0') }}_2.12:1.3.1
+      # need to extract only the major and minor versions
+      - org.apache.iceberg:iceberg-spark-runtime-{{ test_scenario['values']['spark'].split('.')[0] }}.{{ test_scenario['values']['spark'].split('.')[1] }}_2.12:1.4.0
```
Comment (razvan) [inline tests/templates/kuttl/iceberg/10-deploy-spark-app.yaml.j2:49]:
```suggestion
      - org.apache.iceberg:iceberg-spark-runtime-{{ test_scenario['values']['spark'].rsplit('.', maxsplit=1)[0] }}_2.12:1.4.0
```

File: examples/ny-tlc-report-image.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@ spec:
   version: "1.0"
   # everything under /jobs will be copied to /stackable/spark/jobs
   image: docker.stackable.tech/stackable/ny-tlc-report:0.1.0
-  sparkImage: docker.stackable.tech/stackable/pyspark-k8s:3.3.0-stackable0.0.0-dev
+  sparkImage: docker.stackable.tech/stackable/pyspark-k8s:3.5.0-stackable0.0.0-dev
```
Comment (razvan) [inline examples/ny-tlc-report-image.yaml:11]: We don't build PySpark images anymore. Python + PySpark is included in the Spark images.
```suggestion
  sparkImage: docker.stackable.tech/stackable/spark-k8s:3.5.0-stackable0.0.0-dev
```

**Review by razvan — COMMENTED** [review]
Looks good for now. Will wait for the final changes.

File: CHANGELOG.md
Comment (siegfriedweber) [inline CHANGELOG.md:1]:
```diff
- - `vector` `0.26.0` -> `0.31.0` ([#269]).
+ - `vector` `0.26.0` -> `0.33.0` ([#269], [#291]).
```

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/107/

**Review by razvan — APPROVED** [review]

**Pull request merged**